src/cljs/main/broadfcui/page/library/research_purpose.cljs

```diff
@@ -66,25 +59,14 @@
                               (swap! state assoc :disease-checked? new-val))}]
       (when (:disease-checked? @state)
         [:div {:style {:paddingLeft "1.5rem"}}
-         (this :-render-selected-diseases)
-         [OntologyAutosuggest
+         (ontology/render-multiple-ontology-selections
+          {:onClick (fn [id label]  (swap! state update :selected-diseases dissoc id))
```

Comment:
...so that here, this can be `:on-delete (fn [{:keys [id]}] (swap! state update :selected-diseases dissoc id))`